File: subpub_example/src/class_subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

class NumberCounter {

  private:
    ros::Subscriber sub;
  
  public:
    NumberCounter(ros::NodeHandle *nh) {
      sub = nh->subscribe("/counter", 1, &NumberCounter::callback_number, this);
    }
  
    void callback_number(const std_msgs::Int64::ConstPtr& msg) {
      ROS_INFO("Count: [%d]", msg->data);
    }
};

int main (int argc, char **argv)
{
  ros::init(argc, argv, "sub_counter");
  ros::NodeHandle nh;
  NumberCounter nc = NumberCounter(&nh);
  ros::spin();
}

File: src/class_subscriber.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import Float64

class NumberCounter:
  
  def __init__(self):
    self.resultado = 0
    self.sub = rospy.Subscriber("counter", Float64, self.funcion)

  def funcion(self, msg): # Tipo callback
    self.resultado = msg.data  
    print(self.resultado)

if __name__ == "__main__":

  rospy.init_node('subscriber')
  NumberCounter()
  rospy.spin()
